pkg/controller/v1alpha2/core/scopes/healthscope/healthscope.go

```diff
@@ -151,24 +164,32 @@ func CheckDeploymentHealth(ctx context.Context, client client.Client, ref runtim
 		r.Diagnosis = errors.Wrap(err, errHealthCheck).Error()
 		return r
 	}
-	r.Target.UID = deployment.GetUID()
+	r.ComponentName = getComponentNameFromLabel(&deployment)
+	r.TargetWorkload.UID = deployment.GetUID()
+
+	requiredReplicas := int32(0)
+	if deployment.Spec.Replicas != nil {
+		requiredReplicas = *deployment.Spec.Replicas
+	}
+
+	r.Diagnosis = fmt.Sprintf(infoFmtReady, deployment.Status.ReadyReplicas, requiredReplicas)
 
-	if deployment.Status.ReadyReplicas == 0 {
-		r.Diagnosis = fmt.Sprintf(errFmtResourceNotReady, deployment.Status)
+	// Health criteria
+	if deployment.Status.ReadyReplicas != requiredReplicas {
```

Comment:
What is there are more ReadyReplicas?

Reply:
Maybe we can treat this situation as "unhealthy" just like this case ["readyReplicas greater than replicas"](https://github.com/kubernetes/kubernetes/blob/60c421b6f63ae3b3a8ef8c2349a4042697a99496/pkg/apis/apps/validation/validation_test.go#L2229) ?